### PR TITLE
Added support for lemmy shortcut links: /c/community@instance

### PR DESCRIPTION
**Pull request by ktgd**
Lemmy supports lemmy links in a short format "/c/community@instance". To support this, I prefix the currentAccount's instance url to the short link.
An example post with these links: [https://lemm.ee/post/609543](https://lemm.ee/post/609543)

### CODE REVIEW

**Comment by ktgd** [comment]
To support the short links, we need to have the home instance URL.
I don't know that I like having to pass the currentAccount.instance to the openLink() function, but I was unable to get currentAccount inside helpers/LinkHelper.ts due to hook rules.
Is there a better way to handle this?

**Comment by gkasdorf** [comment]
We can do this with `store.getState()` like so

```ts
import store from "./store";

const {accounts} = store.getState();

// Do what we want with the account
console.log(accounts.currentAccount);
```

File: helpers/LinkHelper.ts [hunk 1]
```diff
@@ -64,8 +64,27 @@ export const isPotentialFedSite = (link: string) => {
...
 };
 
-export const isLemmySite = async (link: string) => {
-  const urlComponents = new URL(link);
+export const isLemmySite = async (link: string, instanceUrl: string) => {
+  if (!instanceUrl.startsWith("https://") && !instanceUrl.startsWith("http://")) {
+    instanceUrl = "https://" + instanceUrl;
+  }
+
+  // Handle shortcut links that are formatted: "/c/community@instance". Need to prepend the home instance url
```
Comment (gkasdorf) [inline helpers/LinkHelper.ts:72]: See the comment I left on the PR for how to do this 👍

File: components/ui/markdown/RenderMarkdown.tsx [hunk 2]
```diff
@@ -46,7 +46,7 @@ const RenderMarkdown = ({
   const HEADING_4_SIZE = isSystemTextSize ? 18 / fontScale : 15 + fontModifier;
 
   const onLinkPress = (url): boolean => {
-    openLink(url, navigation);
+    openLink(url, navigation, currentAccount.instance);
```
Comment (gkasdorf) [inline components/ui/markdown/RenderMarkdown.tsx:49]: Don't need to pass in the instance anymore.

File: components/ui/buttons/WebLink.tsx [hunk 3]
```diff
@@ -11,9 +13,10 @@ interface WebLinkProps {
 
 function WebLink({ href, children }: WebLinkProps) {
   const navigation = useNavigation<NativeStackNavigationProp<any>>();
+  const currentAccount = useAppSelector(selectCurrentAccount);
 
   const onPress = () => {
-    openLink(href, navigation);
+    openLink(href, navigation, currentAccount.instance);
```
Comment (gkasdorf) [inline components/ui/buttons/WebLink.tsx:19]: Same here

File: components/ui/markdown/RenderMarkdown.tsx (hunk 2 above)
Comment (ktgd) [reply components/ui/markdown/RenderMarkdown.tsx]: Fixed in the latest commit. Thanks!

**Review by gkasdorf — APPROVED** [review]

**Pull request merged**